src/legacy/core_plugins/console/public/np_ready/application/containers/editor/legacy/console_editor/apply_editor_settings.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { DevToolsSettings } from '../../../../../services';
+import { CoreEditor } from '../../../../../types';
+
+export function applyCurrentSettings(editor: CoreEditor | any, settings: DevToolsSettings) {
```

Comment:
Question: when would `editor` not be a `CoreEditor`? I ask as this definition makes me nervous.

Reply:
This was an oversight - it could be a CustomAceEditor or a CoreEditor (not an "any"). This function is a bit nasty at the moment, but for now I just wanted to keep one place where current user settings are applied across editors.